public/docs/ts/latest/guide/index.jade

```diff
@@ -1,10 +1,501 @@
+  We didn't have a two-way binding in the `HeroListComponent` template; here's an example from a different template:
+
++makeExample('intro/ts/app/hero-detail.component.html')(format=".")
+:marked
+  In two-way binding, a data property value flows to the input box from the component as with property binding.
+  The user's changes also flow back to the component, resetting the property to the latest value,
+  as with event binding.
+  
+  Angular processes *all* data bindings once per JavaScript event cycle, 
+  depth-first from the root of the application component tree. 
+figure
+  img(src="/resources/images/devguide/intro/component-databinding.png" alt="Data Binding" style="float:left; width:300px; margin-left:-40px;margin-right:10px" )
+:marked
+  We don't know all the details yet 
+  but it's clear from these examples that data binding plays an important role in communication 
+  between a template and its component ...
+<br clear="all">  
+figure
+  img(src="/resources/images/devguide/intro/parent-child-binding.png" alt="Parent/Child binding" style="float:left; width:300px; margin-left:-40px;margin-right:10px" )
+:marked
+  ... ***and*** between parent and child components
+<br clear="all">
+
+.l-main-section
+<a id="directive"></a>
+:marked
+  ## The Directive
+figure
+  img(src="/resources/images/devguide/intro/directive.png" alt="Parent child" style="float:left; width:150px; margin-left:-40px;margin-right:10px" )
+:marked
+  Our Angular templates are *dynamic*. When Angular renders them, it transforms the DOM 
+  according to the instructions given by a **Directive**.
+  
+  A directive is a class with directive metadata. In TypeScript we'd apply the `@Directive` decorator
+  to attache metadata to the class.
+<br clear="all">
+:marked
+  We already met one form of directive: the component. A component is a *directive-with-a-template*
+  and the `@Component` decorator is actually a `@Directive` decorator extended with template-oriented features.
+
+.l-sub-section
+  :marked
+    While the **component is technically a directive**, 
+    it is so distictive and central to Angular applications that we chose
+    to separate the component from the directive in our architectural overview.
+:marked
+  There are two *other* kinds of directives as well that we call "structural" and "attribute" directives.
+  
+  **Structural** directives alter layout by adding, removing, and replacing elements in DOM.
+  
+  We see two built-in structural directives at play in our [example](#template) template:
++makeExample('intro/ts/app/hero-list.component.html', 'structural')(format=".")
+:marked
+  * [`*ng-for`](displaying-data.html#ng-for) tells Angular to stamp out one `<div>` per hero in the `heroes` list.
+  * [`*ng-if`](displaying-data.html#ng-if) includes the `HeroDetail` component only if a selected hero exists.
+  
+  **Attribute** directives alter the appearance or behavior of a particular element. 
```

Comment:
Here I also like to point out that attribute directives are applied to **existing** elements. It's a little word that has a massive impact on how the user understands the difference between the directive types.